spec/integration/faces/plugin_spec.rb

```diff
@@ -1,59 +1,20 @@
 require 'spec_helper'
-require 'puppet/face'
-require 'puppet/file_serving/metadata'
-require 'puppet/file_serving/content'
-require 'puppet/indirector/memory'
+require 'webmock/rspec'
 
-module PuppetFaceIntegrationSpecs
 describe "Puppet plugin face" do
-  INDIRECTORS = [
-    Puppet::Indirector::FileMetadata,
-    Puppet::Indirector::FileContent,
-  ]
-
-  INDIRECTED_CLASSES = [
-    Puppet::FileServing::Metadata,
-    Puppet::FileServing::Content,
-    Puppet::Node::Facts,
-  ]
-
-  INDIRECTORS.each do |indirector|
-    class indirector::Memory < Puppet::Indirector::Memory
-      def find(request)
-        model.new('/dev/null', { 'type' => 'directory' })
-      end
-    end
-  end
-
-  before do
-    FileUtils.mkdir(File.join(Puppet[:vardir], 'lib'))
-    FileUtils.mkdir(File.join(Puppet[:vardir], 'facts.d'))
-    FileUtils.mkdir(File.join(Puppet[:vardir], 'locales'))
-    @termini_classes = {}
-    INDIRECTED_CLASSES.each do |indirected|
-      @termini_classes[indirected] = indirected.indirection.terminus_class
-      indirected.indirection.terminus_class = :memory
-    end
-  end
-
-  after do
-    INDIRECTED_CLASSES.each do |indirected|
-      indirected.indirection.terminus_class = @termini_classes[indirected]
-      indirected.indirection.termini.clear
-    end
-  end
+  it "processes a download request" do
+    Puppet[:trace] = true
+    Puppet[:log_level] = 'debug'
 
-  def init_cli_args_and_apply_app(args = ["download"])
-    Puppet::Application.find(:plugin).new(double('command_line', :subcommand_name => :plugin, :args => args))
-  end
+    metadata = "[{\"path\":\"/etc/puppetlabs/code\",\"relative_path\":\".\",\"links\":\"follow\",\"owner\":0,\"group\":0,\"mode\":493,\"checksum\":{\"type\":\"ctime\",\"value\":\"{ctime}2020-07-10 14:00:00 -0700\"},\"type\":\"directory\",\"destination\":null}]"
+    stub_request(:get, %r{/puppet/v3/file_metadatas/(plugins|pluginfacts|locales)}).to_return(status: 200, body: metadata, headers: {'Content-Type' => 'application/json'})
```

Comment:
Specs are failing because the metadata for `facts.d` says the desired mode should be `493 => 0755`, but puppet applies a settings catalog:

```
Notice: /File[/tmp/rspecrun20200716-6590-tb9g62/tmp_settings20200716-6590-1c4qpv2/var/facts.d]/mode: mode changed '0775' to '0755'
```

which results in a plugin "change" instead of the expected "No plugins downloaded"
And that signifies a "change"

Reply:
The `pluginfacts` mount is special because we `use` permissions from the source, whereas `locales` and `plugins` don't. If the local directory is not created with mode `0755`, then a "change" will be reported when the mode is changed to `0755`, which can happen if umask is not `022`. It appears Travis uses `002` so we get the above "change".